docs/Tutorial_DAMP.ipynb

```diff
@@ -0,0 +1,276 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #85.                    chunk_start = max(0, chunk_stop - chunksize)
</pre><p>According to the MATLAB code <code>DAMP_2_0.m</code> , the authors use the following approach to update the chunk (in backward processing).</p><p> </p><p>In the first iteration, they do: </p><pre class="ql-syntax" spellcheck="false">#MATLAB
approximate_distance = min( real(MASS_V2(T(i-X+1:i), query))); 
</pre><p>where X is the initial chunksize.</p><p> </p><p>Then, while <code>approximate_distance &gt; best-so-far-discord-distance</code> , they do: </p><pre class="ql-syntax" spellcheck="false">#MATLAB

# update chunksize
X = 2 * X  # doubling the chunksize
expansion_num = expansion_num + 1   # initially, expansion_num is set to 0 

# update chunk start/end points
X_start = i-X+1+(expansion_num * SubsequenceLength);
X_end = i-(X/2)+(expansion_num * SubsequenceLength);

# compute distance
approximate_distance = min( real(MASS_V2(T(X_start:X_end), query))); 
</pre><p> </p><p>Here, the authors go backward in the time series. They double the chunksize in each iteration. IIUC, the reason for having <code>expansion_num</code> in the code is nothing but just because of the way they update the chunk start, end points using the query index <code>i</code> . </p><p>This is not a clean way! Instead, one can just use the chunk end point and find the chunk start point from it by going backward. This results in a cleaner code compared to the current approach where the authors use the query index to compute the chunk start, end point in each iteration.</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/TDAmeritrade/stumpy/pull/920/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Reply:
<p>If the chunk size is known ahead of time, one should pre-compute the start/stop indices for each chunk using the <code>stumpy.core._get_ranges</code> function</p>

Reply:
@seanlaw 
IIUC, in the original proposed algorithm, the chunksize is getting doubled when the current chunk is checked and it does not result in early-abandoning. So, the next chunk will have the length `2 * len(current_chunk)`.

Reply:
<p>Sure</p>